src/account/AccountManager.ts

```diff
@@ -0,0 +1,241 @@
+
+        const headers = {
+            'Authorization': authToken,
+            'Content-Type': 'application/json'
+        };
+
+        const body = {
+            deviceId: await getDeviceId()
+        };
+
+        try {
+            await AccountManager.executeWithRetry(
+                () => AccountManager.makeRequest(url, body, headers),
+                AccountManager.shouldRetrySignOut,
+                AccountManager.MAX_RETRIES,
+                AccountManager.COMMAND_TIMEOUT,
+                AccountManager.TOTAL_TIMEOUT
+            );
+            console.log(`${AccountManager.LOG_TAG}: Sign out successful`);
+        } catch (error) {
+            console.log(
+                `${AccountManager.LOG_TAG}: Sign out failed with error=`,
+                error instanceof Error ? error.message : 'UNKNOWN'
+            );
+            // Don't throw error for sign out to allow local sign out to proceed
+        }
+    }
+
+    private static async makeRequest(
+        url: string,
+        body: object,
+        headers: Record<string, string>
+    ): Promise<Response> {
+        const controller = new AbortController();
+        const timeoutId = setTimeout(() => controller.abort(), AccountManager.COMMAND_TIMEOUT);
+
+        try {
+            console.log(`${AccountManager.LOG_TAG}: Making request to url=${url}, body=${JSON.stringify(body)}, headers=${JSON.stringify(headers)}`);
+            const response = await fetch(url, {
+                method: 'POST',
+                headers,
+                body: JSON.stringify(body),
+                signal: controller.signal
+            });
+
+            if (!response.ok) {
+                const error = new Error(`HTTP error! status: ${response.status}`) as VizbeeError;
+                error.code = response.status >= 400 && response.status < 500 
+                    ? VizbeeErrorCodes.HTTP_4XX_ERROR 
+                    : 'HTTP_ERROR';
+                throw error;
+            }
+
+            return response;
+        } finally {
+            clearTimeout(timeoutId);
+        }
+    }
+
+    private static shouldRetry(error: Error): boolean {
+        console.log(`${AccountManager.LOG_TAG}: shouldRetry received error=`, error);
+        
+        const vizbeeError = error as VizbeeError;
+        if (vizbeeError.code === VizbeeErrorCodes.HTTP_4XX_ERROR) {
+            return false;
+        }
+        return true;
+    }
+
+    private static shouldRetrySignOut(error: Error): boolean {
+        console.log(`${AccountManager.LOG_TAG}: shouldRetrySignOut received error=`, error);
+        
+        const vizbeeError = error as VizbeeError;
+        if (vizbeeError.code === VizbeeErrorCodes.HTTP_4XX_ERROR) {
+            return false;
+        }
+        // Don't retry on 500 errors for signout
+        if (error.message.includes('status: 500')) {
+            return false;
+        }
+        return true;
+    }
+
+    private static async executeWithRetry(
```

Comment:
`executeWithRetry` should extracted from AccountManager, it should be of a common utility

Reply:
DONE